tests/tests/tier0/bluechi-propogating-service-failure/test_bluechi_propogating_service_failure.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+
+from typing import Dict
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+from bluechi_test.service import Option, Section, SimpleRemainingService
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+NODE_BAR = "node-bar"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+    node_bar = nodes[NODE_BAR]
+
+    service = SimpleRemainingService("target.service")
+    service.set_option(Section.Service, Option.ExecStart, "/bin/false")
+
+    node_bar.install_systemd_service(service)
+
+    service = SimpleRemainingService("requesting.service")
+    service.set_option(Section.Unit, Option.BindTo, "bluechi-proxy@node-bar_target.service")
+    service.set_option(Section.Unit, Option.After, "bluechi-proxy@node-bar_target.service")
+    service.set_option(Section.Service, Option.ExecStart, "/bin/false")
+    service.set_option(Section.Service, Option.RemainAfterExit, "yes")
+
+    node_foo.install_systemd_service(service)
+
+    assert node_foo.systemctl.start_unit("requesting.service")
+    assert node_foo.wait_for_unit_state_to_be("requesting.service", "failed")
+    assert node_bar.wait_for_unit_state_to_be("target.service", "failed")
+
+
+def test_agent_config_c_option(
```

Comment:
Please also rename the directory to `proxy-service-propagate-target-service-failure` and the python file to `test_proxy_service_propagate_target_service_failure`